### PR TITLE
feat(grants-collaboration): expose notes retrieval

### PR DESCRIPTION
**Pull request by greg-adams**
## Description
Created route for notes by grant Id.

Added tests for exposed api route. After discussion with @TylerHendrickson and in an effort to increase data isolation for tests, I updated grants tests to re-seed db between tests. This approach would maintain some seed data for convenience (E.g. users, grants, agencies), but create and rollback records for a particular test. This adds some time to run the tests, but seems OK at least on my machine. I'd like to just rollback records for an individual test, but rollback/transaction approaches don't seem useful here, I'm not sure if a db template would be a faster approach. 

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

File: packages/server/src/routes/grants.js [hunk 1]
```diff
@@ -422,4 +422,20 @@ router.delete('/:grantId/interested/:agencyId', requireUser, async (req, res) =>
...
+    const { user } = req.session;
+    const { paginateFrom, limit } = req.query;
+    const limitInt = limit ? parseInt(limit, 10) : undefined;
+
+    if (limit && (!Number.isInteger(limitInt) || limitInt < 1 || limitInt > 100)) {
+        res.sendStatus(400);
+        return;
+    }
+
+    const rows = await db.getOrganizationNotesForGrant(grantId, user.tenant_id, { afterRevision: paginateFrom, limit: limitInt });
```
[the remark this answers is not in the document]
Comment (greg-adams) [reply packages/server/src/routes/grants.js]: updated!

**Comment by greg-adams** [comment]
Yes - that is the ideal I wanted...sort of a global rollback that we could use after each test to only apply to changes made during the test. Knex does allow for nested transactions, but my read on it appears that nested transactions need to maintain reference to the originating transaction (to be subsequently rolled back). i.e.
```
knex.transaction((trx) => {
  // nested dml uses provided arg instead of a new knex.transaction()
  trx('grants').insert(....)
  
  // ...

  // after test
  trx.rollback();
})
```
I thought through a few approaches (we're not the first to have this issue, see https://github.com/bas080/knest), but some sort of dependency injection seemed too big of a refactor. I could envision a change in the knex.js file where we wrap `knex.transaction()` that returns this global test transaction in tests, or the live one in production but I didn't want to hold up this story.

**Comment by TylerHendrickson** [comment]
> Yes - that is the ideal I wanted...sort of a global rollback that we could use after each test to only apply to changes made during the test. Knex does allow for nested transactions, but my read on it appears that nested transactions need to maintain reference to the originating transaction (to be subsequently rolled back). i.e.
> 
> ```
> knex.transaction((trx) => {
>   // nested dml uses provided arg instead of a new knex.transaction()
>   trx('grants').insert(....)
>   
>   // ...
> 
>   // after test
>   trx.rollback();
> })
> ```
> 
> I thought through a few approaches (we're not the first to have this issue, see https://github.com/bas080/knest), but some sort of dependency injection seemed too big of a refactor. I could envision a change in the exported knex where we wrap `knex.transaction()` that returns this global test transaction in tests, or the live one in production but I didn't want to hold up this story.

Agree with your rationale here – I think your current approach seems good for our needs.

**Review by TylerHendrickson — APPROVED** [review]
@greg-adams Looks good!

**Pull request merged**